bitcoin/src/bip32.rs

```diff
@@ -455,6 +455,8 @@ pub enum Error {
     Base58(base58::Error),
     /// Hexadecimal decoding error
     Hex(hex::Error),
+    /// `PublicKey` hex should be 66 or 130 digits long.
+    InvalidPublicKeyHex(usize),
```

Comment:
Why is this not called `InvalidHexLength` as below? That one is more clear.

Also, we have this problem everywhere: I think all enum variants should use named fields except for inner errors (e.g. `hex::Error` in this case) which are obvious enough.

Reply:
I put `PublicKey` in the name to future proof the varian because the 66/130 in the comments and output are pubkey specific.

Reply:
Can you explain a bit more what you're future-proofing against?

Reply:
I went digging into the code and my brain to try and remember what I was thinking when I wrote both this PR and the comment above. Seems I misspoke in the comment, I believe I chose the name because the error enum in the `bip32` module is more general than just parsing keys, where as in the`key` module the error enum is only related to keys. Moreover the `InvalidHexLength` is only returned when parsing pubkeys. I've changed the `bip32` error variant to `InvalidPublicKeyHexLength` to better show that it is the same as `key::InvalidHexLength` but with pubkey in the variant to disambiguate in like other `bip32::Error` variants.

Reply:
Ah, I think I understand, makes sense.